### ISSUE TITLE
Address Software Dependencies are Affected by Known Vulnerabilities

### ISSUE THREAD

**Issue by vidi42** [hzi-braunschweig/SORMAS-Project#3580]
### Problem Description

The target application uses dependencies that are affected by known vulnerabilities.
Applications that use dependencies which are affected by known vulnerabilities offer an additional attack
surface. Known vulnerabilities could provide a generic path to compromise the application in different ways,
such as extracting sensitive information, manipulating internal application states, or even compromising the
underlying operating system. Depending on the vulnerability and the affected software component, they might
be publicly available proof-of-concept exploits. Such vulnerabilities are very likely to be exploited, especially
on services that are exposed to the Internet where the exploitation could be automated.

The following list of dependencies with known vulnerabilities has been identified:
* jackson-databind:2.0.1 used by com.auth0-java-jwt-2.2.1.jar affected by 35 CVEs
* com.fasterxml.jackson.core-jackson-databind-2.9.8.jar affected by 33 CVEs
* com.google.guava-guava-19.0.jar affected by CVE-2018-10237
* commons-beanutils-commons-beanutils-1.9.2.jar affected by CVE-2019-10086
* commons-beanutils.jar affected by CVE-2019-10086
* org.geotools-gt-shapefile-21.4.jar affected by CVE-2018-18749
* snakeyaml.jar affected by CVE-2017-18640
* vaadin-sass-compiler.jar affected by CVE-2011-0509 and CVE-2019-10799

These libraries were extracted from the release version 1.48.2 of sormas-ui, sormas-ear and sormas-rest.
The current OWASP Dependency Checks also show quite a lot possible vulnarabilities (about 90) in used dependencies (java libraries).

### Proposed Change

1. Check which findings might be solved by a newer library version. Some might already be cleaned by #3449.
2. Assess which findings are not a vulnarability in our setup (effectifly a false positive).
3. Document which findings will not be addressed in a simple version update due to more complex changes or risks.

### System Details
* SORMAS version: 1.48.2

### Additional information

Check this after #3449 and #3584 are done because some version updates there will resolve findings.

**Comment by JonasCir** [comment]
uh not good. I think we should enable GitHub's automatic [dependabot alerts](https://docs.github.com/en/free-pro-team@latest/github/managing-security-vulnerabilities/about-alerts-for-vulnerable-dependencies) for this (and all other) repositories, if possible

**Comment by JonasCir** [comment]
> GitHub sends Dependabot alerts when we detect vulnerabilities affecting your repository.
@fhauptmann and @MartinWahnschaffeSymeda any comments?

**Comment by MateStrysewske** [comment]
@JonasCir Dependabot should already be enabled

**Comment by JonasCir** [comment]
Hm, then I propose to investigate why this was not caught . The CVEs are published in [GitHub Advisory DB](https://github.com/advisories/GHSA-6phf-73q6-gh87) and some of them rated as severe.

**Comment by StefanKock** [comment]
For Dependency Check false positives can be suppressed:
- How suppressions need to be defined as XML file: https://jeremylong.github.io/DependencyCheck/general/suppression.html
- Use `--suppression` as command line argument to supress false positives: https://jeremylong.github.io/DependencyCheck/dependency-check-cli/arguments.html

**Comment by StefanKock** [comment]
Current findings probably not (all) fixed by #3584:
- com.auth0-java-jwt-2.2.1.jar (dependency of com.nexmo:client)
- gradle-wrapper.jar (2.10)
- keycloak-adapter-core.jar
- logback-core-1.1.7.jar
- maven-ant-tasks-2.1.4-SNAPSHOT_PATCHED.jar
- org.geotools-gt-shapefile-21.4.jar
- org.jboss.resteasy-resteasy-jaxrs-3.9.1.Final.jar
- org.keycloak-keycloak-core-11.0.0.jar

Probably fixed with #3584:
- commons-beanutils-1.9.2.jar
- guava-19.0.jar
- hibernate-core-5.4.18.Final.jar
- httpclient-4.5.4.jar

False positives:
- vaadin-sass-compiler.jar CVE-2011-0509 : We are using a newer version than 6.4.9
- vaadin-sass-compiler.jar CVE-2019-10799 : [False positive documented in Dependency Check 6.0.3](https://github.com/jeremylong/DependencyCheck/issues/2511)

**Comment by HolgerReiseVSys** [comment]
`maven-ant-tasks` is used for automating some common tasks during development.
It is not part of the release and is not relied on by the code, tests or the CI pipeline.

According to https://maven.apache.org/ant-tasks/ it has been superseded by [Maven Artifact Resolver Ant Tasks](https://maven.apache.org/resolver-ant-tasks/).

**Comment by StefanKock** [comment]
Evaluation of findings in the comment above, to be discussed with @MartinWahnschaffeSymeda

**Comment by StefanKock** [comment]
Open findings will be addressed in new tickets (especially Keycloak 12 is not available yet).

**Comment by StefanKock** [comment]
Documented all not relevant findings (https://github.com/hzi-braunschweig/SORMAS-Project/issues/3580#issuecomment-740562116) in check-suppressions.xml to exclude them in Dependency-Check on Jenkins, successfully tested.

**Comment by StefanKock** [comment]
I needed to add this argument for Dependency-Check in Jenkins, I forgot that on monday: `--suppression ${WORKSPACE}/sormas-base/dependencies/check-suppression.xml`
